### PR TITLE
move common_koji to module_utils

### PR DESCRIPTION
**Pull request by ktdreyer**
This gets rid of the `PYTHONPATH` hack.

### CODE REVIEW

**Comment by ktdreyer** [comment]
@alfredodeza Ansible has a feature where they merge whatever's in your local `module_utils` directory into their own [module_utils](https://github.com/ansible/ansible/tree/devel/lib/ansible/module_utils). This is the preferable way to share code between modules, but this ends up breaking the tests, because the import paths are different when we run `tox` compared to when we run `ansible-playbook`.

Do you have any suggestions for how to handle this gracefully?

**Comment by ktdreyer** [comment]
One option is to add a second `sys.path` hack to `conftest.py`, and then `try/catch` the `ImportError` like so:

```diff
diff --git a/library/koji_host.py b/library/koji_host.py
index cdb5818..843335b 100644
--- a/library/koji_host.py
+++ b/library/koji_host.py
@@ -2,7 +2,10 @@
 from ansible.module_utils.basic import AnsibleModule
 from ansible.module_utils._text import to_native
 from ansible.errors import AnsibleError
-from ansible.module_utils import common_koji
+try:
+    from ansible.module_utils import common_koji
+except ImportError:
+    import common_koji
 
 
 ANSIBLE_METADATA = {
diff --git a/tests/conftest.py b/tests/conftest.py
index b4770a2..7695833 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -13,3 +13,7 @@ def pytest_sessionstart(session):
     library_path = join(dirname(working_directory), 'library')
     if library_path not in sys.path:
         sys.path.insert(0, library_path)
+
+    module_utils_path = join(dirname(working_directory), 'module_utils')
+    if module_utils_path not in sys.path:
+        sys.path.insert(0, module_utils_path)
```

This allows tox to pass.

I don't really like having the try/catch thing there, though, because it's confusing to read that in the main module code and I'd have to copy & paste it to every module. I wonder if there's any other trick we could do in conftest.py for this.

**Comment by ktdreyer** [comment]
this works in conftest.py:

```python
    location = join(module_utils_path, 'common_koji.py')
    # py35+ needs to use importlib.util instead

    import imp
    foo = imp.load_source("ansible.module_utils.common_koji", location)
    import ansible.module_utils
    ansible.module_utils.common_koji = foo
```

**Comment by ktdreyer** [comment]
@alfredodeza I've pushed something that works for me in Python 2 and Python 3 now.

File: tests/conftest.py [hunk 1]
```diff
@@ -6,10 +6,33 @@ def pytest_sessionstart(session):
     """
     This pytest hook gets executed after the Session object has been created
     and before any collection starts.
-    The main purpose of this hook is to prepend the absolute path of the
-    ``library`` directory so that imports in the test files can work correctly.
+
+    ansible-playbook will automatically load modules from the "library"
+    directory. To mimic this duringt tests, we will prepend the absolute path
```
Comment (alfredodeza) [inline tests/conftest.py]:
```suggestion
    directory. To mimic this during tests, we will prepend the absolute path
```

**Comment by alfredodeza** [comment]
Do you have a source that describes the behavior from Ansible where it merges local module_utils into its own? That sounds like a potential source of pain (how does it deal with name clashes?)

I am OK with this if it is working well for you

**Comment by ktdreyer** [comment]
yeah, docs could be clearer here, so I opened https://github.com/ansible/ansible/pull/51152

**Pull request merged**